src/services/GitHubService.ts

```diff
@@ -0,0 +1,56 @@
+import axios from "axios";
+import GitHubRepository from "../interfaces/GitHubRepository";
+import GitHubPullRequest from "../interfaces/GitHubPullRequest";
+
+class GitHubService {
+	private static instance: GitHubService;
+
+	/* eslint-disable */
+	private constructor() { }
+	/* eslint-enable */
+
+	static getInstance(): GitHubService {
+		if (!this.instance) {
+			this.instance = new GitHubService();
+		}
+
+		return this.instance;
+	}
+
+	async getRepository(user: string, repo: string): Promise<GitHubRepository> {
+		const url = `https://api.github.com/repos/${user}/${repo}`;
+		const { status, data } = await axios.get(url);
+
+		if (status === 200) {
+			return {
+				user: data.owner.login,
+				repo: data.name,
+				description: data.description,
+				language: data.language,
+				url: data.html_url,
+				issues_and_pullrequests_count: data.open_issues_count
+			};
+		} else {
+			throw new Error("There was a problem with the request to GitHub.");
+		}
+	}
+
+	async getPullRequest(user: string, repo: string): Promise<GitHubPullRequest[]> {
+		const url = `https://api.github.com/repos/${user}/${repo}/pulls`;
+		const { data } = await axios.get(url);
+
+		if (data.length !== 0) {
+			const pullRequests = data.map((pull: any) => ({
```

Comment:
It annoys me that you're using `any` here, although I'm not sure how to combat it without creating another interface (which I don't want).

Reply:
Yeah it was giving me errors in VSCode and it gave that as option (and it made it work) so I just ended up using it.